components/script/dom/htmliframeelement.rs

```diff
@@ -335,97 +321,104 @@ impl HTMLIFrameElementLayoutMethods for LayoutJS<HTMLIFrameElement> {
     }
 }
 
-pub trait MozBrowserEventDetailBuilder {
-    #[allow(unsafe_code)]
-    unsafe fn build_mozbrowser_event_detail(&self,
-                                            event: MozBrowserEvent,
-                                            cx: *mut JSContext,
-                                            rval: MutableHandleValue);
+#[allow(unsafe_code)]
+pub unsafe fn build_mozbrowser_custom_event(window: Root<Window>, event: MozBrowserEvent) -> Root<CustomEvent> {
```

Comment:
This function provides a safe abstraction; it should just have an internal `unsafe {}` block.

Also, arguments should *never* be `Root<T>`; use `&Window`.